frontend/src/hooks/useFormLock.tsx

```diff
@@ -0,0 +1,86 @@
+import { useState } from "react";
+import { useAxios } from "../hooks/useAxios";
+
+export const useFormLock = () => {
+  const [formLock, setFormLock] = useState({
+    locked: false,
+    lockedBy: "",
+    lockedTable: "",
+    lockedDate: "",
+    lockedRow: "",
+  });
+
+  const { axiosAll } = useAxios();
+
+  const lockRemover = async (lockInfo: {
+    locked_row_id: string;
+    locked_date: string;
+    locked_table: string;
+    locked_by: string;
+  }) => {
+    await axiosAll().delete(`db_lock/${lockInfo.locked_row_id}`, { headers: lockInfo });
+  };
+  // todo Define a good type. "Any" type temporarily permitted.
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const handleDbLock = async (row: any, rowToLock: string | undefined) => {
```

Comment:
I think it's worth a ticket for making some nice clean types for this kind of function

I dimly recall using something like 'function' as the type, but there's probably something out there in stackOverflow for typescript and async functions. probably it's overkill to go in depth on something like this though.